p2p/conn/connection.go

```diff
@@ -318,13 +340,23 @@ FOR_LOOP:
 			c.Logger.Debug("Send Ping")
 			legacy.WriteOctet(packetTypePing, c.bufWriter, &n, &err)
 			c.sendMonitor.Update(int(n))
+			c.Logger.Debug("Starting pong timer", "dur", c.config.PongTimeout)
+			c.pongTimer = time.AfterFunc(c.config.PongTimeout, func() {
+				c.pongTimeoutCh <- struct{}{}
```

Comment:
Cool! Will this leak ? If the conn stops before we receive this, will there be a stuck go-routine? Maybe we need to drain this channel when we call pongTimer.Stop ?

Reply:
hmm.. it will, most probably

Reply:
https://github.com/tendermint/tendermint/pull/1095/commits/a67865cbd91b3dde2cd54690bcaf77beca6abd01